lib/OpenQA/Log.pm

```diff
@@ -208,8 +208,13 @@ sub setup_log ($app, $logfile = undef, $logdir = undef, $level = undef) {
     OpenQA::App->set_singleton($app);
 }
 
+# same approach as in os-autoinst bmwqemu.pm
 sub redact_settings ($vars) {
-    return {map { $_ !~ qr/(^_SECRET_|_PASSWORD)/ ? ($_ => $vars->{$_}) : ($_ => '[redacted]') } keys %$vars};
+    my $hide_re = '^_SECRET_|_PASSWORD';
+    my $custom_hide_re = eval { qr/$vars->{_HIDE_SECRETS_REGEX}/ } if $vars->{_HIDE_SECRETS_REGEX};
+    $vars->{_HIDE_SECRETS_REGEX} = "(invalid regex specified: $@)" if $@;
+    $hide_re .= "|$custom_hide_re" if $custom_hide_re;
+    return {map { /($hide_re)/ ? ($_ => '[redacted]') : ($_ => $vars->{$_}) } keys %$vars};
```

Comment:
It would probably make sense to compile `$hide_re` into a regex only once outside the `map` call:

```suggestion
    $hide_re = $custom_hide_re ? qr/($hide_re|$custom_hide_re)/ : qr/($hide_re)/;
    return {map { ($_ =~ $hide_re) ? ($_ => '[redacted]') : ($_ => $vars->{$_}) } keys %$vars};
```